### PR TITLE
Minor span tracing improvements

### PR DESCRIPTION
**Pull request by ObsidianMinor**
* [Don't set observer null when ending span](https://github.com/cloudflare/workerd/commit/6e971e5439ec67e4a3f8e2788d3bdb9ff71b80ed)

  Encapsulating the entire span in a state containing the observer and span and setting it null on `end()` prevents the creation of subspans after the span has ended.

* [Add getObserver to SpanBuilder](https://github.com/cloudflare/workerd/commit/e3ec9a9541908a00ddcd793435ca63c0986c2785)

* [Switch span tracing APIs to use kj::ConstString](https://github.com/cloudflare/workerd/commit/66b2b51ca93693f4f6ee5edaec483f5456fe553a)

### CODE REVIEW

File: src/workerd/io/trace.h [hunk 1]
```diff
@@ -428,12 +428,17 @@ class SpanBuilder {
 
 public:
   explicit SpanBuilder(kj::Maybe<kj::Own<SpanObserver>> observer, kj::StringPtr operationName,
-                       kj::Date startTime = kj::systemPreciseCalendarClock().now())
-      { KJ_IF_MAYBE(o, observer) { state.emplace(kj::mv(*o), operationName, startTime); } }
+                       kj::Date startTime = kj::systemPreciseCalendarClock().now()) {
+    if (observer != nullptr) {
+      observer = kj::mv(observer);
```
Comment (kentonv) [inline src/workerd/io/trace.h:433]: This line appears to assign `observer` to itself, so has no effect... do we need more testing?

**Comment by kentonv** [comment]
Why do we need to be able to create subspans after the span has ended? I think the original design expected that would never happen.

**Comment by ObsidianMinor** [comment]
There's nothing saying it's _invalid_, and you can technically do it today by making a separate `SpanParent` to make the child instead of the `SpanBuilder` after it's ended. I guess I intended it more as a consistency thing.

**Comment by ObsidianMinor** [comment]
A real example would probably be something like, operation A spawns operation B, but operation A actually _ends_ before operation B has technically "started" because of runtime shenanigans.

File: WORKSPACE [hunk 2]
```diff
@@ -35,10 +35,10 @@ rules_foreign_cc_dependencies()
 
 http_archive(
     name = "capnp-cpp",
-    sha256 = "d51c3ca9062d4fbd64f66722b08aafe18976aff3e9965aa0958a0afd4323d4b8",
-    strip_prefix = "capnproto-capnproto-b2afb7f/c++",
+    sha256 = "8d4384c4c9c186d56cb25d0224f02b61d0b672118f0b2d67f9ce0c246f201f5e",
+    strip_prefix = "capnproto-capnproto-d0f2270/c++",
     type = "tgz",
-    urls = ["https://github.com/capnproto/capnproto/tarball/b2afb7f8fe393466a38e2fd2ad98482c34aafcee"],
+    urls = ["https://github.com/capnproto/capnproto/tarball/d0f22701fd3e5a95c108ee1bdc0af7f444d67794"],
```
Comment (kentonv) [inline WORKSPACE:41]: Update to final version.

**Review by kentonv — APPROVED** [review]
Looks good other than merge conflicts and updating the capnp dependency.

**Pull request merged**